File: mybatis-plus-join-test/test-join/src/main/java/com/github/yulichang/test/join/entity/AreaDO.java
package com.github.yulichang.test.join.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.experimental.Accessors;

import java.io.Serializable;

@Data
@ToString
@Accessors(chain = true)
@EqualsAndHashCode
@TableName("area")
public class AreaDO implements Serializable {

    @TableId
    private Integer id;

    private String province;

    private String city;

    private String area;

    private String Postcode;

    @TableLogic
    private Boolean del;
}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/entity/AreaDO.kt
package com.github.yulichang.test.kt.entity

import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableLogic
import com.baomidou.mybatisplus.annotation.TableName
import java.io.Serializable

@Suppress("unused")
@TableName("area")
class AreaDO : Serializable {
    @TableId
    var id: Int? = null
    var province: String? = null
    var city: String? = null
    var area: String? = null
    var postcode: String? = null

    @TableLogic
    var del: Boolean? = null
}
